modules/push-notifications/src/providers/AmazonSNS.provider.ts

```diff
@@ -0,0 +1,137 @@
+import AWS from 'aws-sdk';
+import { BaseNotificationProvider } from './base.provider.js';
+import { ConduitGrpcSdk } from '@conduitplatform/grpc-sdk';
+import {
+  ISendNotification,
+  ISendNotificationToManyDevices,
+} from '../interfaces/ISendNotification.js';
+import { IAmazonSNSSettings } from '../interfaces/IAmazonSNSSettings.js';
+
+export class AmazonSNSProvider extends BaseNotificationProvider<IAmazonSNSSettings> {
+  private sns: AWS.SNS;
+  private settings: IAmazonSNSSettings;
+
+  constructor(settings: IAmazonSNSSettings) {
+    super();
+    this._initialized = false;
+    this.isBaseProvider = false;
+    this.updateProvider(settings);
+  }
+
+  updateProvider(settings: IAmazonSNSSettings) {
+    try {
+      this.settings = settings;
+
+      this.sns = new AWS.SNS({
+        credentials: {
+          accessKeyId: settings.accessKeyId,
+          secretAccessKey: settings.secretAccessKey,
+        },
+        region: settings.region,
+      });
+
+      this._initialized = true;
+      ConduitGrpcSdk.Logger.log('Amazon SNS Provider initialized successfully.');
+    } catch (e) {
+      this._initialized = false;
+      ConduitGrpcSdk.Logger.error(
+        `Failed to initialize Amazon SNS: ${(e as Error).message}`,
+      );
+    }
+  }
+
+  private resolvePlatformKey(data?: Record<string, any>): string {
```

Comment:
create a static mapping between AWS platforms and Conduit platform enum. Don't base it on settings.